### DISCUSSION TITLE
Is it normal behavior of tpot connecting tons of udp-ressources of jami/dhtproxy

### DISCUSSION THREAD

**Discussion by KmdKeen** [telekom-security/tpotce#1110] [Q&A]
Hello,

we're kind of concerned that our tpot is rebooting at 5:30 daily via cron and after reboot instantly connects to resources like dhtproxy2.jami.net via UDP and transferring lots of data? We have tons of connections to this network via udp without any reference, what this network is doing for the tpot. 

Best regards,

Sebastian

**Comment by KmdKeen** [comment]
Looks like these sessions are part of the sentrypeer honeypot and are resulting tpot in showing several sentrypeer-alerts promoted by other honeypots... am I right? This is very confusing!

**Reply by ghenry** [reply]
Hi Sebastian,

@t3chn0m4g3 has recently upgraded to v1.4.1 of SentryPeer for TCP mode and I think enabled peer to peer mode (-p) https://github.com/SentryPeer/SentryPeer#command-line-options

This is the DHT which is a public one. SentryPeer uses this to share bad actor hits between nodes.

Thanks,
Gavin.

**Reply by KmdKeen** [reply]
Okay... good to know... we spend some days investigating why an internal deployed tpot is showing SIP alerts from external IPs. 

Right now sentrypeer is startet as a docker container with the following compose code in tpot:

`  sentrypeer:
    container_name: sentrypeer
    restart: always
    networks:
     - sentrypeer_local
    ports:
     - "5060:5060/udp"
    image: "dtagdevsec/sentrypeer:2204"
    read_only: true
    volumes:
     - /data/sentrypeer/log:/var/log/sentrypeer
`

Is there anything we can add here (env/label?) to opt-out of the peer-to-peer part?

**Reply by ghenry** [reply]
It's enabled on the command line here in the Dockerfile https://github.com/telekom-security/tpotce/blob/master/docker/sentrypeer/Dockerfile#L20

You can enable it via env https://github.com/SentryPeer/SentryPeer#environment-variables. 

and you can also run your own private dhtnode to share your own bad actors by setting that via env or cli so no external hits would come in. 

Thanks.

**Reply by t3chn0m4g3** [reply]
@KmdKeen @ghenry Thanks for the discussion. In order to avoid confusion, for the time being, I will disable the peer to peer mode again and will provide an option for opt-in in the docker-compose files based on ENVs. Will ping when the new image is pushed.

**Reply by ghenry** [reply]
No problem. I'm always interested to read about use cases etc. Why did T-Pot reboot @KmdKeen

**Reply by KmdKeen** [reply]
This sounds like a good idea to make it opt-in than opt-out! We were in kind of panic mode for some hours now. ;)

**Reply by KmdKeen** [reply]
> No problem. I'm always interested to read about use cases etc. Why did T-Pot reboot @KmdKeen

It's default behavior of tpot rebooting every day at 5:30 via cron.

**Reply by ghenry** [reply]
Ah, OK. How are you using the SentryPeer data out of interest? :-)

**Reply by KmdKeen** [reply]
@t3chn0m4g3 Just to let you know: I tried setting the ENV but it's not working, as long as the parameter -p is set in the dockerfile. So changing the dockerfile is needed. 

@ghenry We're planning in rolling out tpots all over our network to collect IPs. Sentrypeer was just one of the "default" honeypots enabled after starting the image. We were just very confused when this pot startet showing external source IPs on an internal placed honeypot. ;) 

Plan is to grab the IPs of the different pots to quarantine them on our edge-firewalls/endpoint-firewalls automatically.

**Reply by t3chn0m4g3** [reply]
@KmdKeen Freshly [baked](https://github.com/telekom-security/tpotce/commit/dcf15ca4890b45a2ba562b10ca96e36f1e761af7) and [pushed](https://hub.docker.com/repository/docker/dtagdevsec/sentrypeer).

**Reply by t3chn0m4g3** [reply]
@ghenry Just noticed that something is off with `SENTRYPEER_PEER_TO_PEER=0`, the mere presence of the ENV enables DHT / P2P mode, even with `-p` as CMD parameter disabled. With the ENV not present it works as expected (=false).

Docker output:
```
Attaching to sentrypeer
sentrypeer    | SentryPeer db file location set via cli -f to: /var/log/sentrypeer/sentrypeer.db
sentrypeer    | SentryPeer json log file location set via cli -f to: /var/log/sentrypeer/sentrypeer.json
sentrypeer    | SentryPeer node id: a57db009-1261-4223-8d6c-b51f3d6fbeed
sentrypeer    | Starting sentrypeer...
sentrypeer    | API mode enabled, starting http daemon...
sentrypeer    | SIP mode enabled...
sentrypeer    | Peer to Peer DHT mode enabled...
sentrypeer    | Starting peer to peer DHT mode using OpenDHT-C lib version '2.4.4'...
sentrypeer    | Peer to peer DHT mode started.
sentrypeer    | DHT InfoHash for key 'bad_actors' is: 14d30143330e2e0e922ed4028a60ff96a59800ad
sentrypeer    | Bootstrapping the DHT
sentrypeer    | Waiting 5 seconds for bootstrapping to bootstrap.sentrypeer.org...
sentrypeer    | Configuring local address...
sentrypeer    | Creating sockets...
sentrypeer    | Binding sockets to local address...
sentrypeer    | Listening for incoming UDP connections...
sentrypeer    | SIP responsive mode enabled. Will reply to SIP probes...
sentrypeer    | Listening for incoming TCP connections...
sentrypeer    | Listening for changes to the bad_actors DHT key
```

docker-compose.yml:
```
version: '2.3'

networks:
  sentrypeer_local:

services:

# SentryPeer service
  sentrypeer:
    build: .
    container_name: sentrypeer
    restart: always
#    cpu_count: 1
#    cpus: 0.25
    environment:
#     - SENTRYPEER_WEB_GUI=0
     - SENTRYPEER_PEER_TO_PEER=0
#     - SENTRYPEER_BOOTSTRAP_NODE=bootstrap.sentrypeer.org
     - SENTRYPEER_VERBOSE=1
     - SENTRYPEER_DEBUG=1
    networks:
     - sentrypeer_local
    ports:
     - "4222:4222/udp"
     - "5060:5060/udp"
     - "127.0.0.1:8082:8082"
    image: "dtagdevsec/sentrypeer:2204"
    read_only: true
    volumes:
     - /data/sentrypeer/log:/var/log/sentrypeer
```

Removing ENV from docker-compose => Docker output
```
sentrypeer    | SentryPeer db file location set via cli -f to: /var/log/sentrypeer/sentrypeer.db
sentrypeer    | SentryPeer json log file location set via cli -f to: /var/log/sentrypeer/sentrypeer.json
sentrypeer    | SentryPeer node id: 8be9340a-c782-4171-a483-f3457fbe37db
sentrypeer    | Starting sentrypeer...
sentrypeer    | API mode enabled, starting http daemon...
sentrypeer    | SIP mode enabled...
sentrypeer    | Configuring local address...
sentrypeer    | Creating sockets...
sentrypeer    | Binding sockets to local address...
sentrypeer    | Listening for incoming UDP connections...
sentrypeer    | SIP responsive mode enabled. Will reply to SIP probes...
sentrypeer    | Listening for incoming TCP connections...
```

Maybe I am missing something, but I will comment the ENVs out for [now](https://github.com/telekom-security/tpotce/commit/15f7a1793587b5db2f8acff75f9f3733e80067f9).

@KmdKeen You may want to run `/opt/tpot/update.sh` to account for the changes.

**Reply by ghenry** [reply]
Yeah, I do no truthy tests on the ENV getting. If it's there, it's on for me :-)

https://github.com/SentryPeer/SentryPeer/blob/main/src/conf.c#L266

**Reply by t3chn0m4g3** [reply]
@ghenry I assumed that much 😄. Thanks for the update and clarification.

**Reply by KmdKeen** [reply]
> @KmdKeen You may want to run `/opt/tpot/update.sh` to account for the changes.

Awesome! We're P2P-traffic-free now! Thanks alot!

**Reply by t3chn0m4g3** [reply]
@KmdKeen Awesome and thank you for taking the time for testing / opening the discussion.

**Reply by ghenry** [reply]
Come back soon though!

**Reply by KmdKeen** [reply]
You're welcome! Glad that this was not what we thought at first :D keep up the good work!

**Reply by t3chn0m4g3** [reply]
Yes I bet, that's why I took action right away :)
Thank you!

**Comment by t3chn0m4g3** [comment]
> we're kind of concerned that our tpot is rebooting at 5:30 daily via cron

This is normal behavior, documented [here](https://github.com/telekom-security/tpotce/blob/master/README.md#start-t-pot).

**Reply by KmdKeen** [reply]
Yeah this is fine... we were not concerned about the reboot but the behavior right after ;)

**Reply by t3chn0m4g3** [reply]
Just did not want to leave this part unanswered :bowtie:
